python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,45 +143,272 @@ def img_capabilities():
 
 @image_soda.route("/examples", methods=["GET"])
 def img_examples():
+    """ Get /examples for the service. """
     html = current_app.soda.get_examples(_getparams())
     return make_response(html)
 
 
 @image_soda.route("/tables", methods=["GET"])
 def img_tables():
+    """ Get /tables for the service. """
     xml = current_app.soda.get_tables(_getparams())
     return make_response(xml)
 
 
 @image_soda.route("/sia", methods=["GET"])
 def img_sia():
+    """" Get the /sia service endpoint."""
     resp = current_app.soda.do_sia(_getparams())
     return make_response(resp)
 
 
 @image_soda.route("/sync", methods=["GET", "PUT", "POST"])
 def img_sync():
+    """ Service the /sync request.
+
+    Returns
+    -------
+    resp : `flask.Response`
+        the response.
+    """
     if not request.args:
         soda_url = url_for('api_image_soda.img_sync', _external=True)
         # no parameters present, return service status
         return _service_response(soda_url)
     image = current_app.soda.do_sync(_getparams())
     if image:
-        return _data_response(image)
+        with tempfile.NamedTemporaryFile(prefix="img_", suffix=".fits") as fp:
+            image.writeFits(fp.name)
+            resp = send_file(fp.name,
+                             mimetype="image/fits",
+                             as_attachment=True,
+                             attachment_filename=os.path.basename(fp.name))
+            return resp
     else:
         return _image_not_found()
 
 
 @image_soda.route("/async", methods=["GET", "POST"])
 def img_async():
+    """ Get the /async service endpoint. """
     if not request.args:
         # no parameters present, return service status
         soda_url = url_for('api_image_soda.img_async', _external=True)
         return _service_response(soda_url)
-    xml = current_app.soda.do_async(_getparams())
-    resp = make_response(xml)
-    resp.headers["Content_Type"] = "text/xml"
-    return resp
+    job_id = current_app.soda.do_async(_getparams())
+    return redirect(url_for('api_image_soda.img_async_jobs_job',
+                            job_id=job_id,
+                            _external=True))
+
+
+@image_soda.route("/async-jobs", methods=["GET"])
+def img_async_jobs():
+    """ Get all the jobs of the user in the system. """
+    if session.get("user"):
+        xml = current_app.soda.get_jobs(_getparams())
+        resp = make_response(xml)
+        resp.headers["Content-Type"] = "text/xml"
+        return resp
+    else:
+        raise Exception("Error: user not specified")
+
+
+@image_soda.route("/async-jobs/<job_id>", methods=["GET"])
```

Comment:
I think this should be /async/

Reply:
Great catch - mapping {jobs} to async does make sense, as it simplify things.